File: src/main/java/io/junseok/mealmate/domain/comment/dto/response/CommentInfo.java
package io.junseok.mealmate.domain.comment.dto.response;

import io.junseok.mealmate.domain.comment.entity.Comment;
import java.time.LocalDateTime;
import lombok.Builder;

@Builder
public record CommentInfo(
    String commentContent,
    String email,
    long commentId,
    LocalDateTime createDt,
    LocalDateTime modifyDt
) {
    public static CommentInfo fromEntity(Comment comment){
        return CommentInfo.builder()
            .commentContent(comment.getCommentContent())
            .email(comment.getMember().getEmail())
            .commentId(comment.getCommentId())
            .createDt(comment.getCreateDt())
            .modifyDt(comment.getModifyDt())
            .build();
    }
}

File: src/main/java/io/junseok/mealmate/domain/comment/dto/response/CommentInfo.kt
package io.junseok.mealmate.domain.comment.dto.response;

import io.junseok.mealmate.domain.comment.entity.Comment
import java.time.LocalDateTime

data class CommentInfo(
    val commentContent: String,
    val email: String,
    val commentId: Long,
    val createDt: LocalDateTime,
    val modifyDt: LocalDateTime
)

fun Comment.toCreateCommentInfo() = CommentInfo(
    commentContent = this.commentContent,
    email = this.member.email,
    commentId = this.commentId!!,
    createDt = this.createDt!!,
    modifyDt = this.modifyDt!!
)